src/PhpSpreadsheet/Worksheet/Table.php

```diff
@@ -0,0 +1,471 @@
+<?php
+
+namespace PhpOffice\PhpSpreadsheet\Worksheet;
+
+use PhpOffice\PhpSpreadsheet\Cell\Coordinate;
+use PhpOffice\PhpSpreadsheet\Exception as PhpSpreadsheetException;
+use PhpOffice\PhpSpreadsheet\Worksheet\Table\TableStyle;
+
+class Table
+{
+    /**
+     * Table Name.
+     *
+     * @var string
+     */
+    private $name = '';
+
+    /**
+     * Show Header Row.
+     *
+     * @var bool
+     */
+    private $showHeaderRow = true;
+
+    /**
+     * Show Totals Row.
+     *
+     * @var bool
+     */
+    private $showTotalsRow = false;
+
+    /**
+     * Table Range.
+     *
+     * @var string
+     */
+    private $range = '';
+
+    /**
+     * Table Worksheet.
+     *
+     * @var null|Worksheet
+     */
+    private $workSheet;
+
+    /**
+     * Table Column.
+     *
+     * @var Table\Column[]
+     */
+    private $columns = [];
+
+    /**
+     * Table Style.
+     *
+     * @var TableStyle
+     */
+    private $style;
+
+    /**
+     * Create a new Table.
+     *
+     * @param string $range (e.g. A1:D4)
+     *
+     * @return $this
+     */
+    public function __construct(string $range = '', ?Worksheet $worksheet = null)
+    {
+        $this->setRange($range);
+        $this->setWorksheet($worksheet);
+        $this->style = new TableStyle();
+    }
+
+    /**
+     * Get Table name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Set Table name.
+     *
+     * @return $this
+     */
+    public function setName(string $name)
+    {
+        $name = trim($name);
+
+        if (strlen($name) == 1 && in_array($name, ['C', 'c', 'R', 'r'])) {
+            throw new PhpSpreadsheetException('The table name is invalid');
+        }
+        if (strlen($name) > 255) {
+            throw new PhpSpreadsheetException('The table name cannot be longer than 255 characters');
+        }
+        // Check for A1 or R1C1 cell reference notation
+        if (
+            preg_match(Coordinate::A1_COORDINATE_REGEX, $name) ||
+            preg_match('/^R\[?\-?[0-9]*\]?C\[?\-?[0-9]*\]?$/i', $name)
+        ) {
+            throw new PhpSpreadsheetException('The table name can\'t be the same as a cell reference');
+        }
+        if (!preg_match('/^[A-Z_\\\\]/i', $name)) {
+            throw new PhpSpreadsheetException('The table name must begin a name with a letter, an underscore character (_), or a backslash (\)');
+        }
+        if (!preg_match('/^[A-Z_\\\\][A-Z0-9\._]+$/i', $name)) {
```

Comment:
A literal `.` is also a valid character in a table name

Reply:
Yes, it is already there in the character class ```[A-Z0-9\._]```. I have included a name with a dot in the valid table name test case.